examples/advanced/main.tf

```diff
@@ -32,3 +33,32 @@ module "enterprise" {
     }
   ]
 }
+
+locals {
+  sm_guid   = var.existing_sm_instance_guid == null ? module.secrets_manager[0].secrets_manager_guid : var.existing_sm_instance_guid
+  sm_region = var.existing_sm_instance_region == null ? var.region : var.existing_sm_instance_region
+}
+
+#Create a new SM instance if not using an existing one
+module "secrets_manager" {
+  source               = "terraform-ibm-modules/secrets-manager/ibm"
+  version              = "1.17.8"
+  count                = var.existing_sm_instance_guid == null ? 1 : 0
+  resource_group_id    = null
```

Comment:
You will need to pass a resource group ID, so you should add a resource group module into the example (same as this -> https://github.com/terraform-ibm-modules/terraform-ibm-client-to-site-vpn/blob/4b715db74c0c777296d572ffc40d6837e1a30fdf/examples/ha-complete/main.tf#L10-L20